synapseclient/models/services/storable_entity_components.py

```diff
@@ -203,23 +203,24 @@ async def _store_activity_and_annotations(
 
     if (
         hasattr(root_resource, "activity")
-        and (root_resource.activity or last_persistent_instance)
+        and root_resource.activity is not None
         and (
             last_persistent_instance is None
-            or last_persistent_instance.activity != root_resource.activity
+            or (
```

Comment:
Nit: there is a lot of nested logic going on here, is it possible to break it out into several variables that give insight as to what it is checking for and how it fulfills what the function states that it should be doing?

Reply:
Take a look at this code change and let me know what you think: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1099/commits/8b353053d04619ac5d3039f84244059aece15562

Reply:
Looks good to me! thanks!